### PR TITLE
App url is build on backend not frontend

### LINKED ISSUE #105: Custom app urls are neglected
I just found out that only WAM reverse matches are valid as urls on landing page.
I wrote an PR to change that.
Now at first, reverse match is tried, otherwise given url is used...

Thereby, I stumbled upon an additional `url_arg` which can be passed as config parameter and was used to build the url in template like this:
```
{% if app.url_arg %}
  {% url app.url path=app.url_arg as url %}
{% else %}
  {% url app.url as url %}
{% endif %}
```
Can't rember why? Can't find any app which is using it?!
But I tried to keep this option anyway... (so this should still work with my new suggestion - but couldn't test, because I did not know how?)

### PR DESCRIPTION
**Pull request by henhuy**
Fixes #105
Custom urls could not be rendered otherwise. Now, at first, reverse url match is tried, otherwise url is simply put in place.

### CODE REVIEW

**Comment by Bachibouzouk** [comment]
The test could be to visit the urls of the page and see if we get a 200 response then?

Did you manually test it then?

**Comment by henhuy** [comment]
> The test could be to visit the urls of the page and see if we get a 200 response then?
> 
> Did you manually test it then?

Yes, manually. Unfortunately, it's not so easy to test my app with simple 200 responses, as my stemp tool highly depends on sessions - so testing first page is easy - but testing last simulation page needs a whole set-up session with lots of data and structure. Otherwise there will be only a redirect saying "Session is broken - You have to start new session"...

So, for the moment, as there are only two "real" apps (stemp, stemp_abw) and the changes affect only landing page, I think it is okay to test manually.
But you are totally right - we have to have more test. At least some "smoke tests" (as you suggested) would be fine!

**Comment by henhuy** [comment]
I will checkout this branch on dev and we/ I can check!

**Comment by henhuy** [comment]
Apps seem to work - can you review, so that we can merge?

File: utils/constants.py [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
-
-AppInfo = namedtuple(
-    "AppInfo", ["category", "name", "url", "closed", "icon", "url_arg"],
-)
-AppInfo.__new__.__defaults__ = (False, None, None)
+from django.urls import reverse
+from django.urls.exceptions import NoReverseMatch
+
+
+class AppInfo:
```
Comment (Bachibouzouk) [inline utils/constants.py:6]: A docstring describing the arguments would be good for future references

File: utils/constants.py [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+    def __init__(self, category, name, url, closed=False, icon=None, url_arg=None):
+        self.category = category
+        self.name = name
+        self.__url = url
+        self.closed = closed
+        self.icon = icon
+        self.__url_arg = url_arg
+
+    @property
+    def url(self):
```
Comment (Bachibouzouk) [inline utils/constants.py:42]: A docstring explaining what this function is supposed to do would be helpful :)

**Review by Bachibouzouk — APPROVED** [review]
Also an entry in the changelog would be nice :)

I would only have time to deploy locally tomorrow if you want some otherchecks. The code seems to be doing the job though.

**Pull request merged**